src/solver/impls/petsc/petsc.cxx

```diff
@@ -399,180 +392,40 @@ int PetscSolver::init(int NOUT, BoutReal TIMESTEP) {
     ierr = PetscOptionsHasName(PETSC_NULL,"-J_slowfd",&J_slowfd);CHKERRQ(ierr);
 #endif
     if (J_slowfd) { // create Jacobian matrix by slow fd
-      MatStructure flg;
-
       ierr = SNESSetJacobian(snes,J,J,SNESComputeJacobianDefault,PETSC_NULL);CHKERRQ(ierr);
       output_info << "SNESComputeJacobian J by slow fd...\n";
 
 #if PETSC_VERSION_GE(3,5,0)
       ierr = TSComputeRHSJacobian(ts,simtime,u,J,J);CHKERRQ(ierr);
 #else
+      MatStructure flg;
       ierr = TSComputeRHSJacobian(ts,simtime,u,&J,&J,&flg);CHKERRQ(ierr);
 #endif
       output_info << "compute J by slow fd is done.\n";
       //ierr = MatView(J,PETSC_VIEWER_STDOUT_WORLD);CHKERRQ(ierr);
     } else { // get sparse pattern of the Jacobian
       throw BoutException("Path followed in PETSc solver not yet implemented in general "
                           "-- experimental hard coded values here. Sorry\n");
-
-      output_info << "get sparse pattern of the Jacobian...\n";
-
-      if(n2Dvars() != 0) throw BoutException("PETSc solver can't handle 2D variables yet. Sorry\n");
```

Comment:
I think this throw is related to the earlier comments about not using n2d etc. I think we should probably fix to allow 2D vars, if not we should probably keep this throw and move it to near the start.